core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1578,144 +1409,85 @@ class Log(@volatile private var _dir: File,
         .map(_.messageOffset)
         .getOrElse(maxOffsetInMessages - Integer.MAX_VALUE)
 
-      roll(Some(rollOffset))
+      val newSegment = localLog.roll(Some(rollOffset))
+      afterRoll(newSegment)
+      newSegment
```

Comment:
Could we replace the above 3 lines with `roll(Some(rollOffset))`?

Reply:
Sure.

Reply:
Done in 8ebb39f.